lib/elixir/lib/process.ex

```diff
@@ -834,7 +837,7 @@ defmodule Process do
 
   See `:erlang.process_info/1` for more information.
   """
-  @spec info(pid) :: keyword | nil
+  @spec info(pid) :: [process_info_result_item] | nil
```

Comment:
[nit] Maybe this one is too imprecise here? Since the key will always be an atom, `keyword` might be more precise.

Or should we refer `:erlang.process_info_result_item()` which is a bit more precise? But dialyzer would generalize to `atom` anyway.

Reply:
I don't think this function can actually return individual dictionary keys? 🤔

Reply:
> [nit] Maybe this one is too imprecise here? Since the key will always be an atom, keyword might be more accurate.

> I don't think this function can actually return individual dictionary keys? 🤔

Good points about this function clause, I will revert this change to info/1 spec

> Or should we refer :erlang.process_info_result_item() which is a bit more precise? But dialyzer would generalize to atom anyway.

Unfortunately the type is not exported, Dialyzer says:

```
lib/process.ex:840:30: Unknown type erlang:process_info_result_item/0
```